vxblastsms/ussd.py

```diff
@@ -89,83 +63,98 @@ def handle_raw_inbound_message(self, message_id, request):
-        ussd_session_id = optional_values['ussdSessionId']
-
-        if optional_values['to_addr'] is not None:
+        if values['type'] == '2':  # resume session
             session_event = TransportUserMessage.SESSION_RESUME
-            to_addr = optional_values['to_addr']
-            content = optional_values['request']
+        elif values['type'] == '1':  # new session
+            session_event = TransportUserMessage.SESSION_NEW
         else:
+            # TODO #4: handle type 3 (close session) and type 4 (timeout)
+            # Default to new session for now
             session_event = TransportUserMessage.SESSION_NEW
-            to_addr = optional_values['request']
+
+        if optional_values['msg'] is not None:
+            content = optional_values['msg']
+        else:
             content = None
 
         log.info(
-            'BlastSMSUssdTransport receiving inbound message from %s to %s.' % (
-                from_addr, to_addr))
+            'BlastSMSUssdTransport receiving inbound message from %s to '
+            '%s.' % (values['msisdn'], values['shortcode']))
 
         yield self.publish_message(
             message_id=message_id,
             content=content,
-            to_addr=to_addr,
-            from_addr=from_addr,
+            to_addr=values['shortcode'],
+            from_addr=values['msisdn'],
             session_event=session_event,
             transport_type=self.transport_type,
+            transport_name=self.transport_name,
             transport_metadata={
-                'aat_ussd': {
-                    'provider': provider,
-                    'ussd_session_id': ussd_session_id,
-                }
+                'sessionid': values['sessionid'],
+                'appid': optional_values['appid'],
             },
-            provider=provider,
         )
 
-    def generate_body(self, reply, callback, session_event):
-        request = Element('request')
-        headertext = SubElement(request, 'headertext')
-        headertext.text = reply
+    def generate_body(self, msisdn, sessionid, appid, in_reply_to,
+                      reply_content, session_event):
+
+        e_ussdresp = Element('ussdresp')
 
-        # If this is not a session close event, then send options
+        se_msisdn = SubElement(e_ussdresp, 'msisdn')
+        se_msisdn.text = msisdn
+
+        se_sessionid = SubElement(e_ussdresp, 'sessionid')
+        se_sessionid.text = sessionid
+
+        se_appid = SubElement(e_ussdresp, 'appid')
+        if appid is not None:
+            se_appid.text = appid
+        else:
+            config = self.get_static_config()
+            if config.app_id:
+                se_appid.text = config.app_id
+            else:
+                se_appid.text = None
+
+        # Set Message Type: 2 for Response, 3 for Release
+        se_type = SubElement(e_ussdresp, 'type')
         if session_event != TransportUserMessage.SESSION_CLOSE:
```

Comment:
This will mean that for `SESSION_NEW` we will set the type to `Response`? Just wondering if that's an okay assumption with this API?

Also, maybe we can make these numbers class variables? Then we can use `self.RESPONSE`, or maybe `self.MESSAGE_TYPES['response']`?